PWGCF/FemtoDream/TableProducer/femtoDreamProducerTask.cxx

```diff
@@ -525,7 +774,49 @@ struct femtoDreamProducerTask {
     fillCollisionsAndTracksAndV0<false, true>(col, tracks, fullV0s);
   }
   PROCESS_SWITCH(femtoDreamProducerTask, processData,
-                 "Provide experimental data", true);
+                 "Provide experimental data", false);
+  void
+    processDataCharmHad(aod::FemtoFullCollision const& col,
+                        aod::FemtoHFTracks const& tracks,
+                        soa::Filtered<CandidateLc> const& candidates)
+  {
+    fillCharmHadronTable<false, false>(col, tracks, candidates);
+  }
+  PROCESS_SWITCH(femtoDreamProducerTask, processDataCharmHad,
+                 "Provide experimental data for charm hadron femto", false);
+
+  void
+    processDataCharmHadWithML(aod::FemtoFullCollision const& col,
+                              aod::FemtoHFTracks const& tracks,
+                              soa::Filtered<soa::Join<CandidateLc, aod::HfMlLcToPKPi>> const& candidates)
+  {
+
+    fillCharmHadronTable<false, true>(col, tracks, candidates);
+  }
+  PROCESS_SWITCH(femtoDreamProducerTask, processDataCharmHadWithML,
+                 "Provide experimental data for charm hadron femto with ml", true);
+
+  void processMCCharmHad(aod::FemtoFullCollisionMC const& col,
+                         soa::Join<aod::FemtoHFTracks, aod::McTrackLabels> const& tracks,
+                         soa::Filtered<CandidateLcMC> const& candidates,
+                         GenratedMC const& particles)
+  {
+    fillCharmHadronTable<true, false>(col, tracks, candidates);
+    if (fillLcMCGenTable)
+      fillCharmHadMCGen(col, tracks, particles);
+  }
+  PROCESS_SWITCH(femtoDreamProducerTask, processMCCharmHad, "Provide MC for charm hadron", false);
+
+  void processMCCharmHadWithML(aod::FemtoFullCollisionMC const& col,
+                               soa::Join<aod::FemtoHFTracks, aod::McTrackLabels> const& tracks,
+                               soa::Filtered<soa::Join<CandidateLcMC, aod::HfMlLcToPKPi>> const& candidates,
+                               GenratedMC const& particles)
+  {
+    fillCharmHadronTable<true, true>(col, tracks, candidates);
+    if (fillLcMCGenTable)
+      fillCharmHadMCGen(col, tracks, particles);
+  }
```

Comment:
if my doubt above is correct, then I'd put this piece of code in another separate `process` function, which simply loops over `particles`